custom_components/ocpp/api.py

```diff
@@ -774,6 +777,31 @@ async def _get_specific_response(self, unique_id, timeout):
 
         return resp
 
+    async def measure_connection_latency(self):
+        """Measure the connection latency."""
+        try:
+            while True:
+                t0 = time.perf_counter()
+                pong_waiter = await self._connection.ping()
```

Comment:
use asyncio.wait_for(self._connection.ping(), timeout=20) to ensure the loop does not get stuck if there is no response to the ping